File: Solutions/1694. Reformat Phone Number/1694. Reformat Phone Number.cpp
class Solution {
public:
    string reformatNumber(string number) {
        vector<char> tmp;
        for(auto c:number)
            if(c!='-'&&c!=' ')
                tmp.push_back(c);
        string res;
        for(int i=0;i<tmp.size();){
            if(tmp.size()-i==4){
                res+=tmp[i];
                res+=tmp[i+1];
                i+=2;
            }
            else if(tmp.size()-i==2){
                res+=tmp[i];
                res+=tmp[i+1];
                break;
            }
            else if(tmp.size()-i==3){
                res+=tmp[i];
                res+=tmp[i+1];
                res+=tmp[i+2];
                break;
            }
            else{
                res+=tmp[i];
                res+=tmp[i+1];
                res+=tmp[i+2];
                i+=3;
            }
            res+='-';
        }
        return res;
    }
};

File: Solutions/1694. Reformat Phone Number/1694. Reformat Phone Number.java
class Solution {
    public String reformatNumber(String number) {
        String tmp="";
        for(var c:number.toCharArray())
            if(c!=' '&&c!='-')
                tmp+=c;
        String res="";
        for(int i=0;i<tmp.length();){
            if(tmp.length()-i==4){
                res+=tmp.charAt(i);
                res+=tmp.charAt(i+1);
                i+=2;
            }
            else if(tmp.length()-i==3){
                res+=tmp.charAt(i);
                res+=tmp.charAt(i+1);
                res+=tmp.charAt(i+2);
                break;
            }
            else if(tmp.length()-i==2){
                res+=tmp.charAt(i);
                res+=tmp.charAt(i+1);
                break;
            }
            else{
                res+=tmp.charAt(i);
                res+=tmp.charAt(i+1);
                res+=tmp.charAt(i+2);
                i+=3;
            }
            res+='-';
        }
        return res;
    }
}